moveos/moveos-stdlib/moveos-stdlib/sources/context.move

```diff
@@ -131,15 +121,26 @@ module moveos_std::context {
 
     #[private_generics(T)]
     /// Create a new Object, the owner is the `sender`
-    public fun new_object<T: key>(self: &mut Context, value: T): Object<T> {
+    /// Add the Object to the global object storage and return the ObjectRef
+    public fun new_object<T: key>(self: &mut Context, value: T): ObjectRef<T> {
+        let id = fresh_object_id(self);
         let owner = sender(self);
-        object::new<T>(&mut self.tx_context, owner, value)
+        new_object_with_id(self, id, owner, value)
     }
 
     #[private_generics(T)]
     /// Create a new Object with owner
-    public fun new_object_with_owner<T: key>(self: &mut Context, owner: address, value: T): Object<T> {
-        object::new(&mut self.tx_context, owner, value)
+    /// Add the Object to the global object storage and return the ObjectRef
+    public fun new_object_with_owner<T: key>(self: &mut Context, owner: address, value: T): ObjectRef<T> {
+        let object_id = fresh_object_id(self);
+        new_object_with_id(self, object_id, owner, value)
+    }
+
+    public(friend) fun new_object_with_id<T: key>(self: &mut Context, id: ObjectID, owner: address, value: T) : ObjectRef<T> {
```

Comment:
If new an object and return `ObjectRef `by default, need to additionally call the `object_ref::id` method to get the object_id. Looking at the example,  only need the object_id in many cases, which will increase the burden to the developer.

Reply:
Previously, developers needed to call `object::id(&obj)` to get ObjectID, now, replaced by `object_ref::id(&obj_ref)`.